source/extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <iostream>
+#include <memory>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+
+#include "quiche/quic/core/quic_buffer_allocator.h"
+#include "quiche/quic/platform/api/quic_logging.h"
+
+namespace quic {
+
+// Implements the interface required by
+// https://quiche.googlesource.com/quiche/+/refs/heads/master/quic/platform/api/quic_mem_slice.h
+class QuicMemSliceImpl {
+public:
+  // Constructs an empty QuicMemSliceImpl.
+  QuicMemSliceImpl() = default;
+
+  // A helper function to construct a buffer fragment and the buffer of given length the fragment
+  // points to with the guarantee that buffer and fragment are both aligned according to
+  // max_align_t.
+  static Envoy::Buffer::BufferFragmentImpl*
+  allocateBufferAndFragment(QuicBufferAllocator* allocator, size_t length) {
+    auto bundle = reinterpret_cast<BufferFragmentBundle*>(
+        allocator->New(sizeof(BufferFragmentBundle) + length));
+    Envoy::Buffer::BufferFragmentImpl& fragment = bundle->fragment_with_padding.fragment;
+    return new (&fragment) Envoy::Buffer::BufferFragmentImpl(
+        bundle->buffer, length,
+        [allocator, bundle](const void*, size_t, const Envoy::Buffer::BufferFragmentImpl*) {
```

Comment:
std::function created from lambdas with large captures also tends to allocate memory. Since the BufferFragmentImpl pointer is already given in the 3rd argument, let's just delete that instead and do not capture |bundle|.

Reply:
done